docs/user/BuiltInFunctions.md

```diff
@@ -392,10 +392,10 @@ EXISTS(missing)     -- false
 
 ### EXTRACT
 
-Given a date part and a timestamp returns then timestamp's date part value. 
+Given a date part and a datetime type returns then datetime's date part value. 
 
 Signature
-: `EXTRACT: ExtractDatePart Timestamp -> Integer`
+: `EXTRACT: ExtractDatePart DateTime -> Integer`
```

Comment:
It does. But for the `double`, the output value has `e0` suffixed. I wasn't sure if that would be a breaking issue for some of the existing customers as the corresponding tests needed to be altered.
For now, I have changed this to `double`.

Reply:
A couple of points here.

1. changing the return type of this function may break existing customers who expect it to be an integer.
1. While postgres [appears to always return `double precision` from the `extract` function](https://www.postgresql.org/docs/9.1/functions-datetime.html#FUNCTIONS-DATETIME-EXTRACT), the SQL-92 spec requires an "exact numeric" (i.e. `decimal`) with implementation-defined precision and a scale of 0.  

Given the above points, some offline discussion with the team may be warranted.  Compatibility with `postgres` is a nice goal, but I think the SQL-92 spec may take precedence here.  Backward compatibility with our existing customers is also a concern.

SQL-92 specification, section 6.6.4:

```
    4) If <extract expression> is specified, then

            Case:

            a) If <datetime field> does not specify SECOND, then the data
              type of the result is exact numeric with implementation-
              defined precision and scale 0.

            b) Otherwise, the data type of the result is exact numeric
              with implementation-defined precision and scale. The
              implementation-defined scale shall not be less than the spec-
              ified or implied <time fractional seconds precision> or <in-
              terval fractional seconds precision>, as appropriate, of the
              SECOND <datetime field> of the <extract source>.
```

Definition of "exact numeric":

```
         4.4.1  Characteristics of numbers

         An exact numeric value has a precision and a scale. The precision
         is a positive integer that determines the number of significant
         digits in a particular radix (binary or decimal). The scale is a
         non-negative integer. A scale of 0 indicates that the number is an
         integer. For a scale of S, the exact numeric value is the integer
         value of the significant digits multiplied by 10-S.
```